draft-irtf-cfrg-opaque.md

```diff
@@ -967,93 +1102,273 @@ enc_server_info
 : Optional application-specific information to exchange during the protocol encrypted
 under key Ke2, defined below.
 
-mac
+server_mac
 : An authentication tag computed over the handshake transcript computed using Km2,
 defined below.
 
 ~~~
 struct {
-  uint8 mac[Nm];
+  uint8 client_mac[Nm];
 } KE3;
 ~~~
 
-mac
+client_mac
 : An authentication tag computed over the handshake transcript computed using
-Km3, defined below.
+Km2, defined below.
+
+#### OPAQUE-3DH Functions {#opaque-3dh-functions}
+
+##### OPAQUE-3DH Message Protocol functions
+
+~~~
+AKEStart(credential_request, client_info)
 
-#### OPAQUE-3DH Key Schedule {#derive-3dh}
+Parameters:
+- Nn, the nonce length
+
+Input:
+- credential_request, a CredentialRequest structure
+- client_info, the optional client_info to send to the server. Note that it will not be encrypted, and only authenticated 
 
-OPAQUE-3DH requires MAC keys `server_mac_key` and `client_mac_key` and
-encryption key `handshake_encrypt_key`. Additionally, OPAQUE-3DH also
-outputs `session_key`. The schedule for computing this key material is below.
+Output:
+- ke1, a KE1 structure
+- client_secret, the client's ephemeral secret key of the Diffie-Hellman key exchange
 
+Steps:
+1. client_nonce = random(Nn)
+2. client_secret = RandomPrivateKey()
+3. client_keyshare = RecoverPublicKey(ephemeral_client_secret_key)
```

Comment:
`ephemeral_client_secret_key` is undefined?

Reply:
it's `client_secret`, of course 🤦